paracelsus/graph.py

```diff
@@ -48,5 +52,62 @@ def get_graph_string(
         raise ValueError(f"Unknown Format: {format}")
     transformer = transformers[format]
 
+    # Keep only the tables which were included / not-excluded
+    include_tables, exclude_tables = resolve_include_exclude_tables(
+        include_tables=include_tables, exclude_tables=exclude_tables, all_tables=set(metadata.tables.keys())
+    )
+    check_included_table_fk(metadata=metadata, include_tables=include_tables)
+    filtered_metadata = filter_metadata(metadata=metadata, include_tables=include_tables)
+
     # Save the graph structure to string.
-    return str(transformer(metadata))
+    return str(transformer(filtered_metadata))
+
+
+def resolve_include_exclude_tables(
+    exclude_tables: Set[str],
+    include_tables: Set[str],
+    all_tables: Set[str],
+) -> tuple[Set[str], Set[str]]:
+    """Given sets of tables to include/exclude and the set of all tables, resolve the ones to include/exclude."""
+    match len(include_tables), len(exclude_tables):
+        case 0, 0:
+            return all_tables, set()
+        case 0, int():
+            return all_tables - exclude_tables, exclude_tables
+        case int(), 0:
+            if not include_tables.issubset(all_tables):
+                non_existent_tables = include_tables - all_tables
+                raise ValueError(
+                    f"Some tables to include ({non_existent_tables}) don't exist within the found tables ({all_tables})."
+                )
+            return include_tables, all_tables - include_tables
+        case _:
+            raise ValueError(
+                f"Only one or none of include_tables ({include_tables}) or exclude_tables ({exclude_tables}) can contain values."
+            )
+
+
+def check_included_table_fk(
+    metadata: MetaData,
+    include_tables: Set[str],
+):
+    """Check that all foreign keys in the included_tables set point to included tables."""
+    for table_name in include_tables:
+        for fk in metadata.tables[table_name].foreign_keys:
+            if fk.column.table.name not in include_tables:
+                raise RuntimeError(
+                    f"Table '{table_name}' contains a foreign key to '{fk.column.table.name}' "
+                    "which is excluded / not-included."
+                )
```

Comment:
If the table has a foreign key to an excluded table I think we should still create a graph (and include the foreign key as a column of that graph), we should just not include the connection to the other table.

For instance, if we use the example graph from the readme but excluded the`users` table it should generate this graph:

```mermaid
erDiagram
 posts {
  CHAR(32) id PK
  CHAR(32) author FK
  TEXT content "nullable"
  DATETIME created
  BOOLEAN live "nullable"
 }

 comments {
  CHAR(32) id PK
  CHAR(32) author FK
  CHAR(32) post FK "nullable"
  TEXT content "nullable"
  DATETIME created
  BOOLEAN live "nullable"
 }

 posts ||--o{ comments : post
```

Reply:
I was planning on not touching on the `transformers` package but this change requires it (not sure if there's another way to do this). So I removed the error and I'm skipping the connection when building the representations directly (if the project grows maybe having a common representation and then passing them to the transformer classes could be useful).

I think this also makes sense when a referenced model (table) was not imported and registered in `metadata`, I guess currently an unhandled error is raised inside the transformers.